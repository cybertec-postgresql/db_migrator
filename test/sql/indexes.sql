\i test/psql.sql

SELECT plan(4);

SELECT results_eq(
    $$ SELECT statement FROM extschema.construct_indexes_statements(plugin => 'noop_migrator') $$,
    ARRAY[
        'CREATE INDEX t1_idx1_nonunique ON sch1.t1 (c1 ASC)',
        'CREATE UNIQUE INDEX t1_idx2_unique ON sch1.t1 (c1 ASC)',
        'CREATE INDEX t2_idx1_filtered ON sch1.t2 (c1 ASC) WHERE (c1 = 1)'
    ],
    'Statements for index creation should be correct'
);

SELECT is(
    extschema.db_migrate_indexes(plugin => 'noop_migrator'),
    0,
    'Should create user-defined indexes without errors'
);

SELECT indexes_are(
    'sch1',
    't1',
    ARRAY['t1_idx1_nonunique', 't1_idx2_unique']
);

SELECT index_is_unique(
    'sch1',
    't1',
    't1_idx2_unique'
);

SELECT * FROM finish();
