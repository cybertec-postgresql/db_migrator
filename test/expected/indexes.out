\i test/psql.sql
\unset ECHO
1..4
ok 1 - Statements for index creation should be correct
ok 2 - Should create user-defined indexes without errors
ok 3 - Table sch1.t1 should have the correct indexes
ok 4 - Index t1_idx2_unique should be unique
